src/main/java/net/minecraftforge/common/ForgeConfig.java

```diff
@@ -73,7 +80,30 @@ public static class Server {
             permissionHandler = builder
                     .comment("The permission handler used by the server. Defaults to forge:default_handler if no such handler with that name is registered.")
                     .translation("forge.configgui.permissionHandler")
-                    .define("permissionHandler", "forge:default_handler");
+                    .worldRestart()
+                    .useConfigGuiWidgetFactory(() -> ConfigGuiWidget.TextWidget.FACTORY)
+                    .withErrorDescriber((value) -> {
+                        if (value instanceof String name) {
+                            PermissionGatherEvent.Handler handlerEvent = new PermissionGatherEvent.Handler();
+                            MinecraftForge.EVENT_BUS.post(handlerEvent);
+                            if (!handlerEvent.getAvailablePermissionHandlerFactories().containsKey(new ResourceLocation(name))) {
+                                return Component.translatable("forge.configgui.error.permissionHandler.unknownName", name, handlerEvent.getAvailablePermissionHandlerFactories().keySet().stream().map(Object::toString).collect(Collectors.joining("\n - ", "\n - ", "")));
+                            };
+
+                            throw new IllegalStateException("Passed in valid name for error describer");
+                        }
+
+                        return Component.translatable("forge.configgui.error.permissionHandler.notText");
+                    })
+                    .define("permissionHandler", "forge:default_handler", obj -> {
+                        if (obj instanceof String name) {
+                            PermissionGatherEvent.Handler handlerEvent = new PermissionGatherEvent.Handler();
+                            MinecraftForge.EVENT_BUS.post(handlerEvent);
```

Comment:
Firing an event in a validator sort of seems wrong. Like I get why you are doing it... it just feels wrong though and like the available factories should just be cached.